nixos/modules/tasks/filesystems/zfs.nix

```diff
@@ -335,19 +353,24 @@ in
                   ;;
               esac
             done
-            ''] ++ (map (pool: ''
+          ''] ++ (map (pool: (importLib {
+            zpoolCmd = "zpool";
+            awkCmd = "awk";
+            inherit pool cfgZfs;
+          }) + ''
             echo -n "importing root ZFS pool \"${pool}\"..."
-            trial=0
-            until msg="$(zpool import -d ${cfgZfs.devNodes} -N $ZFS_FORCE '${pool}' 2>&1)"; do
-              sleep 0.25
+            # Loop across the import until it succeeds, because the devices needed may not be discovered yet.
+            imported && exit
```

Comment:
This is not an independent script. I don't think `exit` is a good choice here.

Reply:
Fixed.